### PR TITLE
ci: add reviewers based on label

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by dundargoc** [comment]
Adding @glacambre as reviewer because javascript

**Comment by dundargoc** [comment]
Closed on accident.

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -32,3 +32,39 @@ jobs:
...
+  add-reviewer:
+    runs-on: ubuntu-latest
+    needs: ["triage", "type-scope"]
+    steps:
+      - uses: actions/github-script@v5
+        with:
+          script: |
+            const reviewers = []
+
+            const { data: { labels: labels } } = await github.rest.pulls.get({
```
Comment (glacambre) [inline .github/workflows/labeler.yml:45]: This should work too:
```suggestion
            const { data: { labels } } = await github.rest.pulls.get({
```

**Review by glacambre — APPROVED** [review]
I don't have a good understanding of the ci-specific stuff but LGTM :)

**Review by glacambre — APPROVED** [review]

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -32,3 +32,39 @@ jobs:
...
+      - uses: actions/github-script@v5
+        with:
+          script: |
+            const reviewers = []
+
+            const { data: { labels } } = await github.rest.pulls.get({
+             owner: context.repo.owner,
+             repo: context.repo.repo,
+             pull_number: context.issue.number
+             })
```
Comment (zeertzjq) [inline .github/workflows/labeler.yml:49]: strange indent

Comment (dundargoc) [reply .github/workflows/labeler.yml:49]: Thanks

**Comment by zeertzjq** [comment]
Does this also request review on drafts?

**Comment by dundargoc** [comment]
> Does this also request review on drafts?

It requests reviews when a PR is opened, draft or not. We can fine tune it once we know it works as intended.

**Review by jamessan — APPROVED** [review]

**Comment by zeertzjq** [comment]
Looks like it is still not working: <https://github.com/neovim/neovim/runs/5104382834?check_suite_focus=true>

**Comment by dundargoc** [comment]
> Looks like it is still not working: [neovim/neovim/runs/5104382834?check_suite_focus=true](https://github.com/neovim/neovim/runs/5104382834?check_suite_focus=true)

[One fine day I won't forget about the write permissions](https://github.com/neovim/neovim/pull/17330)

**Pull request merged**